Proposed changes Sectioned Shapes (#93)
bSI-InfraRoom/IFC-Specification: IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricModelResource/Entities/IfcSectionedSolidHorizontal/Documentation.md

An _IfcSectionedSolidHorizontal_ is a solid model constructed by sweeping potentially varying cross sections along a curve horizontally.

> The solid is generated by sweeping the _CrossSections_ between _CrossSectionPositions_ with linear interpolation of profile points.  The profile normal is derived from the associated _IfcAxis2PlacementLinear_, not necessarily the tangent of the _Directrix_. The profile X axis is oriented perpendicularly to the left of the _Directrix_ (same direction as _RefDirection_ derived in _IfcAxis2PlacementLinear_) as facing forward along the directrix, and the profile Y axis is oriented upwards according to _Axis_.

Reviewer: Simplification.
```suggestion
The solid is generated by sweeping the _CrossSections_ between _CrossSectionPositions_ with linear interpolation between profile points with the same tag along the directrix.  The profile normal is derived from the associated _IfcAxis2PlacementLinear_, not necessarily the tangent of the _Directrix_. The profile X axis is the direction of _RefDirection_ from _IfcAxis2PlacementLinear_ as facing forward along the directrix, and the profile Y axis is the direction of _Axis_.
```

Reviewer 2: I'd propose a correction of the sentence "The profile X axis is the direction of _RefDirection_ from _IfcAxis2PlacementLinear_". Considering https://github.com/bSI-InfraRoom/IFC-Specification/issues/49 , the profile X axis should be equal to the direction of the implicit **_IfcAxis2PlacementLinear Y axis_**. And I'd also want to make clear that the 2d profile is actually defined in the YZ plane of _IfcAxis2PlacementLinear_ (unlike other usages of IfcProfileDef, when it's always defined in the XY plane of the underlying coordinate system). Hope that makes sense.

Reviewer 3: like to ping @SergejMuhic, @pjanck and @jmirtsch - is it really agreed to leave the all-time long specification that all profiles or cross sections in a sweeping generation are swept along the local z-axis and that the profile is defined in local x/y? Haven't followed all recent discussions but this sounds very new to me.

Reviewer 4: I'm not quite following here. Here is my understanding of the above statements:

- `IfcProfileDef` is defined in its own, local *(x,y)* coordinate system and in its view, it is getting extruded along the local *z* axis (which is what @TLiebich is saying)
- each solid that uses profiles then defines how the profile's local coordinate system gets mapped to the solid's coordinate system (which is the question @Nadia-Sugar is asking)

Therefore, confirming Nadia's statement: in my view, the profile's *(x,y)* gets mapped to the `IfcAxis2PlacementLinear`'s *(y,z)* in this case. **Which makes the statement above wrong**, it should read:

> The profile *normal* is the (corrected) direction of _RefDirection_ from _IfcAxis2PlacementLinear_ as facing forward along the directrix, and the profile Y axis is the (corrected) direction of _Axis_ from _IfcAxis2PlacementLinear_.

or 

> The profile X axis is the resulting Y axis from _IfcAxis2PlacementLinear_ as facing forward along the directrix, and the profile Y axis is the direction of _Axis_ from _IfcAxis2PlacementLinear_.

From the two, I prefer the former.

Reviewer: I will review `IfcProfileDef` documentation.

Reviewer 3: in two other occasions 'IfcSurfaceCurveSweptAreaSolid' and 'IfcFixedReferenceSweptAreaSolid' the mapping of (x,y) of the 2D profile coordinates into the 3D Cartesian transformation operator along the directrix is precisely defined:

The orientation of the SweptArea as it sweeps along the Directrix is precisely defined by a CartesianTransformationOperator3d with attributes:

- LocalOrigin as point (0; 0; 0),
- Axis1 as the orthogonal projection of the FixedReference onto the normal plane defined by Axis3.
- Axis3 as the direction of the tangent vector t at the point of the Directrix with parameter u.
- The remaining attributes are defaulted to define a corresponding transformation matrix T(u), which varies with the Directrix parameter u.

and profile x = Axis 1, and profile y = Axis 2

Reviewer 3: in e.g. 'IfcSectionedSolidHorizontal' the implicit positioning through an CartesianTransformationOperator3d is replaced by an explicit positioning via 'IfcAxis2PlacementLinear' - but the mapping to the Axis 1, 2, 3 should be the same.

- LocalOrigin as point (0; 0; 0), positioned along the directrix by 'IfcAxis2PlacementLinear.Location' being the 'IfcPointByDistanceExpression'
- Axis 1 as the orthogonal projection of the 'IfcAxis2PlacementLinear.RefDirection' onto the normal plane defined by Axis3.
- Axis 3 as the as the direction of the IfcAxis2PlacementLinear.Axis'
- Axis 2 is defaulted

Reviewer: > and profile x = Axis 1, and profile y = Axis 2

Where is this from? In principle Axis1, Axis2 and Axis3 are merely mappings for the underlying coordinate system. Axis1 being X specifically is an agreement. I do not see it stated in profile definitions either.

Author: Here's the concept when introduced in IFC4.1 https://standards.buildingsmart.org/IFC/RELEASE/IFC4_1/FINAL/HTML/link/ifcsectionedsolidhorizontal.htm
The documentation nominates profile x as lateral to directrix, and y as a "vertical" direction (Z).  The revised concept for 4.3 is similar.
It reuses a common concept IfcAxis2PlacementLinear.  Having the Axis in a Vertical sense is consistent with IfcAxis2Placement3D and is more important in my opinion.  I don't believe replicating the placement linear concept makes sense.  It's not ideal that the horizontal sweep is not consistent with others, but to me it's the best choice.

Reviewer 3: agree - in this example of IfcSectionedSolidHorizontal it is stated:

... where the profile normal agrees with the tangent of the Directrix, the profile X axis is oriented perpendicularly to the left of the Directrix (same direction as LateralOffset at IfcDistanceExpression) as facing forward along the directrix, and the profile Y axis is oriented upwards according to FixedAxisVertical.

so any profile would be positioned in the x= left when looking forward along the directrix, and y = vertical plane with profile.Axis1 being x, and profile.Axis2 being y.
